examples/mass_delete_old_medias.py

```diff
@@ -0,0 +1,562 @@
+            add_date = datetime.strptime(media['add_date'], '%Y-%m-%d %H:%M:%S').date()
+            categories = {cat.strip(' \r\t').lower() for cat in (media['categories'] or '').split('\n')}
+            if add_date < before_date and (not skip_category or skip_category not in categories):
+                old_medias.append(media)
+            else:
+                media_pp = f'{media["title"]} [{media["oid"]}]'
+                if add_date >= before_date:
+                    before_date_pp = before_date.strftime('%Y-%m-%d')
+                    logger.debug(
+                        f'{media_pp} was skipped because it was added after {before_date_pp}.'
+                    )
+                else:
+                    logger.debug(
+                        f'{media_pp} was skipped because it has the skip_category.'
+                    )
+
+    storage_used = sum(media['storage_used'] for media in old_medias)
+    logger.info(
+        f'Found {len(old_medias)} medias added before {before_date.strftime("%Y-%m-%d")} '
+        f'(size: {pp_size(storage_used)}).'
+    )
+    return old_medias
+
+
+def _get_users(msc: MediaServerClient, page_size=500):
+    users = []
+    offset = 0
+    response = msc.api('users/', params={'limit': page_size, 'offset': offset})
+    while response['users']:
+        users += response['users']
+        offset += page_size
+        response = msc.api('users/', params={'limit': page_size, 'offset': offset})
+    return users
+
+
+def _prepare_mail(
+    msc: MediaServerClient,
+    sender: str,
+    speaker_email: str,
+    medias: list[dict],
+    delete_date: date,
+    skip_category: str,
+    html_template: Optional[str],
+    plain_template: Optional[str],
+    email_subject_template: str,
+) -> tuple[str, dict]:
+    # Ensure each media is only once in the list.
+    medias = list({media['oid']: media for media in medias}.values())
+
+    ms_url = msc.conf['SERVER_URL'] + '/edit/'
+    context = {
+        'media_count': len(medias),
+        'media_size_pp': pp_size(sum(media['storage_used'] for media in medias)),
+        'delete_date': delete_date.strftime('%B %d, %Y'),
+        'skip_category': skip_category,
+    }
+    message = MIMEMultipart('alternative')
+    message['Subject'] = email_subject_template.format(**context)
+    message['From'] = sender
+    message['To'] = speaker_email
+
+    if plain_template:
+        plain_media_list = '\n'.join(
+            f'\t- {ms_url}{media["oid"]}/#id_categories - "{media["title"]}" - '
+            f'viewed {media["views_last_year"]} times last year, '
+            f'{media["views_last_month"]} times last month'
+            for media in medias
+        )
+        plain = plain_template.format(list_of_media=plain_media_list, **context)
+        message.attach(MIMEText(plain, 'plain'))
+    if html_template:
+        html_media_list = '\n'.join(
+            f'<li><a href="{ms_url}{media["oid"]}/#id_categories">"{media["title"]}"</a> '
```

Comment:
When testing i realize we have to add another link on the line for easy preview, for example:

["Réduction des déchets & Green IT - Quelles compatibilités entre technologies et développement durable ?"](https://ubicast.tv/edit/v1245d4614d706b64be4/) 0 times last month (click ["here"](https://ubicast.tv/edit/v1245d4614d706b64be4/#id_categories) to protect against deletion)